sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/implementation/HttpTraceUtil.java

```diff
@@ -121,5 +125,29 @@ public static Status parseResponseStatus(int statusCode, Throwable error) {
                 return Status.UNKNOWN.withDescription(message);
         }
     }
+
+    /**
+     * Parse the resource provider name from the user span name.
+     *
+     * @param userSpanName the span name populated in the RestProxy layer.
+     *
+     * @return the resource provider group name.
+     */
+    public static String parseNamespaceProvider(String userSpanName) {
+        if (userSpanName.contains("KeyVault")) {
+            return "Microsoft.KeyVault";
+        }
+        if (userSpanName.contains("AppConfig")) {
+            return "Microsoft.AppConfiguration";
+        }
+        if (userSpanName.contains("Storage")) {
+            return "Microsoft.Storage";
+        }
+        if (userSpanName.contains("TextAnalytics")) {
+            return "Microsoft.CognitiveServices";
+        }
```

Comment:
How reliably can we determine whether the `SwaggerInterfaceParser`'s `serviceName` will contain the expected names here? Instead within `RestProxy` could we retrieve the name of the package where the method exists and pass through an additional context value which will be a bit more robust compared to either a codegen name/hand-written name?

Reply:
I agree it does not seem like a reliable/easy to maintain way to determine resource provider namespace. Could each library set some common global variable? It should also be possible to fall back to default value (null ?) and don't set az.namespace attribute in this case eliminating libraries that don't support tracing

Reply:
I agree, this is fraught with danger. Let's find a better solution.

Reply:
Updated the approach to allow the client libraries to send in their respective `Resource Provider Names` to pass through the context object and skip the hand-written comparisons. Also, updated it to skip adding the attribute if az.namespace value isn't set.